src/cr/cube/cube.py

```diff
@@ -131,20 +138,20 @@ def population_fraction(self):
         return self._cubes[0].population_fraction
 
     @lazyproperty
-    def n_responses(self):
+    def n_responses(self) -> int:
         """Total number of responses considered from first cube in this set."""
         return self._cubes[0].n_responses
 
     @lazyproperty
-    def valid_counts_summary(self):
+    def valid_counts_summary(self) -> int:
         """The valid count summary values from first cube in this set."""
         return self._cubes[0].valid_counts_summary
 
     @lazyproperty
-    def _cubes(self):
+    def _cubes(self) -> Tuple["Cube", ...]:
         """Sequence of Cube objects containing data for this analysis."""
 
-        def iter_cubes():
+        def iter_cubes() -> Iterator[Cube]:
```

Comment:
I'm wondering why here you're using `Cube` and 3 lines above `"Cube"`

Reply:
Because this one is defined inside of the function, which means each time the parent function is called, so it's aware of the definition of the `Cube` class. The parent function, on the other hand, is defined when the file is loaded for the first time, and is not aware of the `Cube` class at the time of the definition.